src/west/app/config.py

```diff
@@ -182,20 +194,27 @@ def check_config(self, option):
             self.die(f'invalid configuration option "{option}"; '
                      'expected "section.key" format')
 
+    def _read(self, name, configfile):
+        value = self.config.get(name, configfile=configfile)
+        if value is None:
+            self.dbg(f'{name} is unset')
+            raise CommandError(returncode=1)
+
     def read(self, args):
         self.check_config(args.name)
-        value = self.config.get(args.name, configfile=args.configfile or ALL)
-        if value is not None:
-            self.inf(value)
-        else:
-            self.dbg(f'{args.name} is unset')
-            raise CommandError(returncode=1)
+        value = self._read(args.name, args.configfile or ALL)
+        self.inf(value)
 
     def write(self, args):
         self.check_config(args.name)
         what = args.configfile or LOCAL
+        if args.append:
+            value = self._read(args.name, args.configfile or ALL) or ''
```

Comment:
This would raise an exception if not found, is this desired?

Reply:
Yes, it is caught and you get an exit code (though no error message)